### PR TITLE
Initial work of moving out attention code from Agent class

### PR DESCRIPTION
**Pull request by misgeatgit**

### CODE REVIEW

File: opencog/atoms/base/Atom.cc [hunk 1]
```diff
@@ -201,6 +201,17 @@ void Atom::setAttentionValue(AttentionValuePtr av)
     avch(getHandle(), local, av);
 }
 
+void Atom::stimulate(AtomSpace* as, float stimulus)
```
Comment (linas) [inline opencog/atoms/base/Atom.cc]: this can't work, atoms must not depend on the atomspace ...

Comment (linas) [inline opencog/atoms/base/Atom.cc]: The solution that makes sense is to add a method to AtomSpace something like 
```
AtomSpace::stimulate(const Handle& h, double stimulus)
```
that does what your code does, but flipping things around.

Also -- please don't use float -- use double.  floats are just 32 bit, with 23-bit mantissas and will plague you with rounding errors.

Comment (misgeatgit) [inline opencog/atoms/base/Atom.cc]: I have made those changes.

**Comment by linas** [comment]
thanks, looks great!

**Comment by linas** [comment]
this pertains to bug opencog/opencog#2314

**Pull request merged**